### PR TITLE
Add test for preference API integration with Authentication & Recovery portal

### PR DESCRIPTION
**Pull request by inthirakumaaran**
**Purpose**
Test cases for Integration of preference API to  Authentication & Recovery portal. Related Issue https://github.com/wso2/product-is/issues/10935

**Related PRs**

Depends on https://github.com/wso2/identity-apps/pull/1486,  https://github.com/wso2/carbon-identity-framework/pull/3317

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.identity.integration.common.clients.Idp.IdentityProviderMgtServiceClient;
+import org.wso2.identity.integration.test.oauth2.OAuth2ServiceAbstractIntegrationTest;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.IOException;
+
+/**
+ * This test class is to test Preference API integration with Login & Recovery Portals.
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java:42]:
```suggestion
 * Tests Preference API integration with Login & Recovery Portals.
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    private static final String RECOVERY_ENDPOINT_NOTIFICATION_CONTENT = "name=\"recoveryOption\" value=\"EMAIL\"";
+    private static final String CREATE_ACCOUNT_CONTENT = "id=\"registerLink\"";
+    private static final String RECOVERY_ENDPOINT_URL =
+            "https://localhost:9853/accountrecoveryendpoint/recoveraccountrouter.do";
+
+    private IdentityProvider superTenantResidentIDP;
+    private IdentityProviderMgtServiceClient superTenantIDPMgtClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java:64]: Using a data provider can't we run the test for a tenant too?

Comment (inthirakumaaran) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java:89]: Yes we can add a tenant using a data provider

**Review by mefarazath — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        superTenantIDPMgtClient = new IdentityProviderMgtServiceClient(sessionCookie, backendURL);
+        superTenantResidentIDP = superTenantIDPMgtClient.getResidentIdP();
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+        recoveryEndpoint = "https://localhost:9853/t/" + activeTenant + RECOVERY_ENDPOINT_URL;
```
Comment (IsuraD) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java:95]: The host and port shouldn't be hardcoded.  You may use backendServerURL variable or UrlGenerationUtil.getBackendURL util

**Pull request merged**